suite-native/module-add-accounts/src/screens/AddCoinDiscoveryRunningScreen.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { networks, NetworkSymbol } from '@suite-common/wallet-config';
+import { VStack, Text, Spinner, SpinnerLoadingState } from '@suite-native/atoms';
+import { Screen } from '@suite-native/navigation';
+import { Translation } from '@suite-native/intl';
+import { Account } from '@suite-common/wallet-types';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    selectDeviceAccountsByNetworkSymbol,
+    selectDeviceDiscovery,
+} from '@suite-common/wallet-core';
+import { toggleEnabledDiscoveryNetworkSymbol } from '@suite-native/discovery';
+import { useCoinEnabling } from '@suite-native/coin-enabling';
+
+import { useAddCoinAccount } from '../hooks/useAddCoinAccount';
+
+export type AddCoinDiscoveryRunningScreenProps = {
+    networkSymbol: NetworkSymbol;
+};
+
+export const AddCoinDiscoveryRunningScreen = ({ route }) => {
+    const { networkSymbol, flowType } = route.params;
+    const dispatch = useDispatch();
+    const accounts = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccountsByNetworkSymbol(state, networkSymbol),
+    );
+    const normalAccounts = accounts.filter(a => a.accountType === 'normal');
+    const nonEmptyAccounts = accounts.filter(a => !a.empty);
+
+    const discovery = useSelector(selectDeviceDiscovery);
+    const { navigateToAddCoinDiscoveryFinishedScreen, navigateToSuccessorScreen } =
+        useAddCoinAccount();
+    const { enabledNetworkSymbols, applyDiscoveryChanges } = useCoinEnabling();
+    const [loadingResult, setLoadingResult] = useState<SpinnerLoadingState>('idle');
+
+    const goToAccountDetail = ({ account }: { account: Account }) => {
+        navigateToSuccessorScreen({
```

Comment:
what is the benefit of calling this navigation helper here rather calling naigation.replace("AccountDetail", ...)? Does not the helper function just add an extra step. IMO it decreases readability, because you need to go to the `navigateToSuccessorScreen` and check its definition. IF I'm correct, it should always resolve to navigation to an account detail, right?

Reply:
It should either go to receive or account detail (different UI), depending on from where the flow started: `flowType`

Reply:
the `flowType` is accessible is also accessible in this screen no?

Reply:
Hmm. Good point. I'll thing about it once more. Thanks

Reply:
there is a last  use of navigateToSuccessorScreen and that one is in the hook. I can replace it with actual content of that function, but to me it felt it makes the code less readable, because it adds 20 lines of code. The other navigation helpers were already removed